### PR TITLE
Clean up generated code

### LINKED ISSUE #13: Search, fix and remove all `TODO(cookiecutter)` comments in the code
### What happened?

The repository was generated using repo-config, but many `TODO(cookiecutter)` comments were left in the files.

### What did you expect instead?

There should be no `TODO`s in the files and things that needed to be done, should be done :)

### Affected version(s)

_No response_

### Affected part(s)

Documentation (part:docs), Unit, integration and performance tests (part:tests), Build script, CI, dependencies, etc. (part:tooling)

### Extra information

_No response_

### PR DESCRIPTION
**Pull request by jack-herrmann**
Hopefully the requested changes to the structure were performed correctly

### CODE REVIEW

**Comment by jack-herrmann** [comment]
There are three commits because I worked on top of the first two, because they haven't been merged yet

File: src/py.typed
Comment (llucax) [inline src/py.typed:1]: So this should be `src/pylint_datetime/py.typed`.

File: src/datetime_time_checker.py
Comment (llucax) [inline src/datetime_time_checker.py:1]: So this should be `src/pylint_datetime/__init__.py`.

File: tests/test_pylint_datetime.py [hunk 1]
```diff
@@ -4,7 +4,23 @@
...
+
+    Returns:
+        True if no exception was raised.
+
+    Raises:
+        RuntimeError: if blow_up is True.
+    """
+    if blow_up:
+        raise RuntimeError("This function should be removed!")
+    return True
```
Comment (llucax) [inline tests/test_pylint_datetime.py]: Can you just remove all the code from this file, or maybe just leaving an empty test as a place holder?

**Review by llucax — CHANGES_REQUESTED** [review]
Oh, it seems we can't just ship one module and mark it as typed. It seems like [the only way to specify that a single module is typed is by just putting it into a package](https://github.com/python/typing/issues/1333#issuecomment-1376532053), so it seems we need the directory and a `__init__.py` after all.

I would also split this `Structure changes to repo` into 3 commits:

1. Only formatting changes.
2. Removal of files that shouldn't be stored in the repo
3. Fixing `tests/test_pylint_datetime.py` (ideally removing the tests for `delete_me` instead of adding the `delete_me` function

I would also rebase this on the current `v0.x.x` so it is the first PR we merge, so the other PRs don't have garbage on them (updating binary files that shouldn't be there).

**Review by llucax — DISMISSED** [review]

**Review by llucax — APPROVED** [review]

**Pull request merged**